polar-core/src/data_filtering.rs

```diff
@@ -201,19 +201,18 @@ impl VarInfo {
 
     /// turn dot expressions into symbols but leave other things unchanged.
     fn undot(&mut self, term: &Term) -> Value {
-        let val = term.value();
-        match val.as_expression() {
+        match term.as_expression() {
             Ok(Operation {
                 operator: Operator::Dot,
                 args,
             }) if args.len() == 2 => Value::from(self.dot_var(&args[0], &args[1])),
-            _ => val.clone(),
+            _ => term.value().clone(),
```

Comment:
This was the lone instance where moving the `.as_*()` methods from `Value` -> `Term` resulted in more work (`.as_expression()` calls `Term::value` and then we call `Term::value` a second time on this line)